pkg/log/handler.go

```diff
@@ -0,0 +1,45 @@
+package log
+
+import (
+	"fmt"
+	"github.com/applike/gosoline/pkg/cfg"
+	"time"
+)
+
+type Handler interface {
+	Channels() []string
+	Level() int
+	Log(timestamp time.Time, level int, msg string, args []interface{}, err error, data Data) error
+}
+
+type HandlerFactory func(config cfg.Config, handlerIndex int) (Handler, error)
+
+var handlerFactories = map[string]HandlerFactory{
+	"iowriter": handlerIoWriterFactory,
+}
+
+func AddHandlerFactory(name string, factory HandlerFactory) {
+	handlerFactories[name] = factory
```

Comment:
Also, why isn't the handlerIoWriterFactory using the `AddHandlerFactory` method?

Reply:
changed